src/Misc/layoutbin/RunnerService.js

```diff
@@ -19,12 +19,16 @@ var listener = null;
 var runService = function() {
     var listenerExePath = path.join(__dirname, '../bin/Runner.Listener');
     var interactive = process.argv[2] === "interactive";
+    var runonce = process.argv[2] === "once";
 
     if(!stopping) {
         try {
             if (interactive) {
                 console.log('Starting Runner listener interactively');
                 listener = childProcess.spawn(listenerExePath, ['run'], { env: process.env });
+            } else if {
+                console.log('Staring runner listener in RunOnce mode');
+                listener = childProcess.spawn(listenerExePath, ['run', '--once'], { env: process.env });
```

Comment:
At a glance it looked like this is the only place that cares about the startup type at the moment: https://github.com/actions/runner/blob/6922f3cb8684bf1151884422b246687232804aaf/src/Runner.Listener/Runner.cs#L358

So I didn't send `--startuptype service` here, and made these modes exclusive (you can either be interactive, `--once` (and interactive) or `--startuptype service`). LMK if that doesn't make sense!